presto-hive-metastore/src/main/java/com/facebook/presto/hive/HiveType.java

```diff
@@ -201,19 +201,19 @@ public static HiveType toHiveType(TypeInfo typeInfo)
         return new HiveType(typeInfo);
     }
 
-    public static HiveType toHiveType(
-            Type type)
+    public static HiveType toHiveType(Type type, boolean useUnboundedVarchar)
     {
-        return toHiveType(type, Optional.empty());
+        return toHiveType(type, Optional.empty(), useUnboundedVarchar);
```

Comment:
I haven't read through the PR yet. But `useUnboundedVarchar` sounds a very specialized flag that will pollute a lot of generic interfaces this PR touches. Do we have generic ways to handle it. Meanwhile, I will read the PR to understand deeper of the stack.